### PR TITLE
org.graalvm.js:js was renamed to org.graalvm.polyglot:js-community

### PR DESCRIPTION
**Pull request by gsmet**
This is a follow-up of https://github.com/quarkusio/quarkus/pull/39260 .

### CODE REVIEW

**Comment by gsmet** [comment]
@jerboaa interested in your feedback too.

**Review by zakkak — APPROVED** [review]
LGTM

**Comment by jerboaa** [comment]
Seems in-line with https://github.com/oracle/graaljs/blob/master/CHANGELOG.md#version-2310. It's not clear for what that is used, though. It might potentially break 23.0 users, but that's speculation.

Perhaps it would make sense to decouple Graal SDK version and graal js? Similar to truffle-api, perhaps? Then again I'm not sure what those deps are used for.

**Comment by jerboaa** [comment]
Also the enterprise artefact seems to be just `js` over `js-community`, which seems to suggest that quarkus users wouldn't be able to use them interchangeably? Was the previously supported?

**Comment by gsmet** [comment]
Yeah the fact that hey have both seems unfortunate.
Maybe we should declare both.

Also I think the parent first rule I added won't work as this particular artifact is just a pom and we need to declare the actual impl as parent first.

Let me do another run...

**Comment by gsmet** [comment]
@jerboaa @zakkak I adjusted a bunch of things. Not exactly excited about having a ton of stuff parent first but that's apparently what's required now.

James confirmed that it was working for him.

File: bom/application/pom.xml [hunk 1]
```diff
@@ -4424,7 +4424,13 @@
                 <version>${graal-sdk.version}</version>
             </dependency>
             <dependency>
-                <groupId>org.graalvm.js</groupId>
+                <groupId>org.graalvm.polyglot</groupId>
+                <artifactId>js-community</artifactId>
+                <version>${graal-sdk.version}</version>
+            </dependency>
+            <!-- Same but for Gradle Enterprise -->
```
Comment (jerboaa) [inline bom/application/pom.xml:4431]: `s/Gradle/Graal VM/`

File: core/runtime/pom.xml [hunk 2]
```diff
@@ -157,11 +157,22 @@
...
-                        <parentFirstArtifact>com.ibm.icu:icu4j</parentFirstArtifact> <!-- dependency of org.graalvm.js:js -->
+                        <parentFirstArtifact>org.graalvm.truffle:truffle-runtime</parentFirstArtifact>
+                        <parentFirstArtifact>org.graalvm.truffle:truffle-compiler</parentFirstArtifact>
+                        <parentFirstArtifact>org.graalvm.truffle:truffle-enterprise</parentFirstArtifact>
+                        <parentFirstArtifact>org.graalvm.shadowed:icu4j</parentFirstArtifact>
+                        <parentFirstArtifact>org.graalvm.sdk:jniutils</parentFirstArtifact>
+                        <parentFirstArtifact>org.graalvm.sdk:word</parentFirstArtifact>
+                        <parentFirstArtifact>org.graalvm.sdk:native-image</parentFirstArtifact>
+                        <parentFirstArtifact>org.graalvm.sdk:native-bridge</parentFirstArtifact>
+                        <!-- support for GraalVM js -->
```
Comment (jerboaa) [inline core/runtime/pom.xml:175]: Nit `/support` :)

**Review by jerboaa — APPROVED** [review]
Seems fine to me. A few nits.

File: bom/application/pom.xml (hunk 1 above)
Comment (gsmet) [reply bom/application/pom.xml:4431]: Raaah, I have been working with Gradle Enterprise/Develocity for too long...

**Comment by gsmet** [comment]
@zakkak as for testing, probably better if you talk with James, I have no experience with this stuff.

**Comment by jamesnetherton** [comment]
> @zakkak as for testing, probably better if you talk with James, I have no experience with this stuff.

If we're talking specifically about testing the JS bits in native mode. Then from my camel-quarkus POV, I'm not overly bothered about whether it works or not. The js-dsl extension was added primarily for camel-k and it only supports native builds via the Mandrel builder image, where JS is not supported.

**Comment by zakkak** [comment]
@jamesnetherton I am mostly interested to see if that breaks native image builds of projects not actually using the JS-bits, but that somehow end up fetching the corresponding dependencies. 

We have already got https://github.com/quarkusio/quarkus/issues/39322 with the update of `graal-sdk` from 23.0 to 23.1 so I would like to somehow test that this PR won't cause something similar.

**Review by jerboaa — APPROVED** [review]

**Pull request merged**